homeassistant/components/sonos/switch.py

```diff
@@ -357,6 +360,43 @@ async def async_handle_switch_on_off(self, turn_on: bool) -> None:
             _LOGGER.error("Could not update %s: %s", self.entity_id, exc)
 
 
+@callback
+def async_migrate_alarm_unique_ids(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    household_id: str,
+    alarm_ids: list[str],
+) -> None:
+    """Migrate alarm switch unique_ids in the entity registry to the new format."""
```

Comment:
Is there a way we can tell if they have already been migrated and return early without iterating them all?

Reply:
I suppose we could do something like bump the config entry version, but I'm not sure that would be best practice.

Reply:
Config entry version number is a good idea

Reply:
Per discussion in discord, since this only happens when they are added (not always) and its only temporary, it not going to be a problem to let it check every time.